gen_wiki.py

```diff
@@ -0,0 +1,614 @@
+
+    @staticmethod
+    def items(multi_page: bool = False) -> str:
+        """The function to add all items to the wiki.
+
+        Arguments
+        ---------
+        - multi_page (boolean): if the item info should be made for a multi-page wiki or not.
+
+        Returns
+        -------
+        A markdown string that contains information about all items.
+        """
+        out = f"""#{'' if multi_page else '#'} Items
+Those are all items present in the game, that can be traded or found.
+"""
+
+        for item in sorted(items):
+            out += '\n'
+            out += Wiki.item_info(item=item, multi_page=multi_page)
+        return out
+
+    @staticmethod
+    def item_info(item: str, multi_page: bool = False) -> str:
+        """The function to collect information and attributes of a specific item
+
+        Arguments
+        ---------
+        - item (string): The item to collect the information of.
+        - multi_page (boolean): if the item info should be made for a multi-page wiki or not.
+
+        Returns
+        -------
+        A markdown string with the information about the item.
+        """
+        return f"""##{'' if multi_page else '#'} {items[item]["pretty_name"]}
+{items[item]["desc"]}
+
+- Price: {items[item]["price"]}
+- Can be used in fights: {"Yes" if items[item]["fn"] is not None else "No"}
+"""
+
+    @staticmethod
+    def effects(multi_page: bool = False) -> str:
+        """The function to add all effects to the wiki.
+
+        Arguments
+        ---------
+        - multi_page (boolean): if the item info should be made for a multi-page wiki or not.
+
+        Returns
+        -------
+        A markdown string of all the effects in the game.
+        """
+        if multi_page:
+            out = """# Effects
+Those effects can be given to a Pokete through an attack.
+"""
+            out += str.join("", [effect.ret_md() for effect in effect_list]).replace("###", "##")
+        else:
+            out = """
+## Effects
+Those effects can be given to a Pokete through an attack.
+"""
+            out += str.join("", [effect.ret_md() for effect in effect_list])
+        return out
+
+    @staticmethod
+    def single(filename: str = "wiki.md") -> None:
+        """The function to generate a single page wiki.
+
+        This function creates the pokete wiki in a single file and adds the following to it:
+        - title
+        - table of contents
+        - all poketes with information on them
+        - all attacks with information on them
+        - all types with information on them
+        - all items with information on them
+        - all effects with information on them
+
+        Arguments:
+        ---------
+        - filename (string): The file to save the wiki to.
+        """
+        if quiet or verbose:
+            print(":: Generating wiki.md...")
+        if quiet or verbose:
+            print("==> Adding page start...")
+        md_str = Wiki.start()
+        if quiet or verbose:
```

Comment:
why don't just loop over two lists? This would remove a ton of duplicate code.
```suggestion
        for _text, obj in zip(["table of contents", "poketes", "attacks", "types", "items", "effects"],
                              [Wiki.table_of_contents, Wiki.poketes, Wiki.attacks, Wiki.types, Wiki.items, Wiki.effects]):  
            if quiet or verbose:
                print(f"==> Adding {_text}...")
            md_str += obj()
```

Reply:
See b7784883b9619ee676147292cacc14e5590d61ef